addons/stock/models/stock_inventory.py

```diff
@@ -200,13 +200,48 @@ def action_cancel_draft(self):
     @api.multi
     def action_start(self):
         for inventory in self:
-            vals = {'state': 'confirm', 'date': fields.Datetime.now()}
             if (inventory.filter != 'partial') and not inventory.line_ids:
-                vals.update({'line_ids': [(0, 0, line_values) for line_values in inventory._get_inventory_lines_values()]})
+                line_values_list = inventory._get_inventory_lines_values()
+                line_ids = []
+                for line_values in line_values_list:
+                    line_id = inventory._insert_inventory_line(line_values)
+                    line_ids.append(line_id)
+                lines = self.env['stock.inventory.line'].browse(line_ids)
+                lines.modified(self.env['stock.inventory.line']._fields)
+
+            vals = {'state': 'confirm', 'date': fields.Datetime.now()}
             inventory.write(vals)
+        self.recompute()
         return True
+
     prepare_inventory = action_start
 
+    def _insert_inventory_line(self, values):
+        row = self._inventory_line_values_to_row(values)
+        cols = ['id', 'create_uid', 'create_date', 'write_uid', 'write_date'] + list(row)
+        fmts = ["nextval('stock_inventory_line_id_seq')", "%s", "now() AT TIME ZONE 'UTC'",
+                "%s", "now() AT TIME ZONE 'UTC'"] + ["%s"] * len(row)
+        params = (self.env.uid, self.env.uid) + tuple(row.values())
+        query = 'INSERT INTO "stock_inventory_line" ({}) VALUES ({}) RETURNING id'
+        self.env.cr.execute(
+            query.format(", ".join('"%s"' % col for col in cols), ", ".join(fmts)),
+            params,
+        )
+        return self.env.cr.fetchone()[0]
+
+    def _inventory_line_values_to_row(self, values):
+        return {
+            "inventory_id": self.id,
+            "product_id": values['product_id'] or None,
+            "product_uom_id": values.get('product_uom_id') or None,
+            "prod_lot_id": values.get('prod_lot_id') or None,
+            "package_id": values.get('package_id') or None,
+            "theoretical_qty": values.get('theoretical_qty', 0),
```

Comment:
`theoretical_qty` is a computed field
@rco-odoo

Reply:
@fmdl you are right. This line is pointless.